### ISSUE TITLE
\Illuminate\Auth\Guard::logout() does not clear session data

### ISSUE THREAD

**Issue by tormit** [laravel/framework#8661]
Currently it only clears login information, but all data saved during session, must be cleared after user has been signed out.

Only option now is to clear and regenerate session manually.

**Comment by lukasgeiter** [comment]
That's intentional. Not all session data has to be directly related to authentication. Imagine things like a language preference or just flash messages stored in the session. You wouldn't want to clear those just because the user logs out.

So yes, if you need your session cleared then you have to do it yourself.

**Comment by tormit** [comment]
Then there should be some convenient way to store secured data in session, which is always destroyed after logout.

Something like
```php
   Session::secureSet($key,$val);
```
or some special class
```php
   class SecureStore extends \Illuminate\Session\Store {
      // specifies special bag, where to save destructible data
   }
```

**Comment by crynobone** [comment]
```php
Event::listen('user.logout', function () {
    // flush anything that related to logged in user.
});
```

Could also create an event handler class and register it to `user.logout` in `App\Providers\EventServiceProvider`.

**Comment by juukie** [comment]
Great tip @crynobone :)

**Comment by progmars** [comment]
It would be great to warn about this moment in http://laravel.com/docs/5.0/session to avoid having security hole when one user has access to other user's data for session, if he logs in from the same browser without clearing old session cookies. Developers should clearly know that they are responsible for removing sensitive data. I know many developers who intuitively associate "session" with "authentication", and for them this behavior is a surprise.

Even better would be to have ability to specify some "$authSpecific" flag when setting session variables and make Laravel to expire these variables when authentication changes or expires.

**Comment by kanian** [comment]
@promars I agree with you 100%.

**Comment by mpmurph** [comment]
I just came across this unexpected loophole in my app - it was causing data to be incorrectly persist to a user signing up after another user had logged out. I found [this discussion](http://stackoverflow.com/questions/30728741/laravel-session-data-survives-log-out-log-in-even-for-different-users) on stackoverflow and, as a result, have overwritten getLogout() to include Session::flush(). While that erases the session data for the user who is logging out, it doesn't erase their session file - I am not clear what triggers that? The flush() method solves my issue for user's who are purposely logging out but I am still not sure if it is resolved for users who simply abandon an open browser (a concern also raised by the person who posed the stackoverflow question). Will have to do some more testing.

I upvote @progmars' suggestion that this be mentioned in Laravel's Session docs - and it would be great to have a secureSet() method as suggested by @tormit - or the $authSpecific flag @progmars proposes - something to isolate data you want specifically associated with a given authorized user (and no one else).
